src/Cowsay.zig

```diff
@@ -42,8 +36,12 @@ pub fn think(self: *Self, comptime fmt: []const u8, comptime args: anytype) !voi
 }
 
 fn print(self: *Self, comptime fmt: []const u8, comptime args: anytype) !void {
-    var buffer = std.ArrayList(u8).init(self.allocator);
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
```

Comment:
Why do you use ArenaAllocator not GeneralPurposeAllocator? What is the difference?
I would like to allow users to specify their own allocator if they want, meanwhile does not want to clutter the function call arguments, that is why I put allocator as a field of the struct. Is that very un-ziggy?

Reply:
It is totally fine to allow users to specify their own allocator!  However, it is important to notice that there is no allocated state that must be kept between function runs within the struct. Instead, the allocator is only necessary for a temporary allocation within the function.

Thus, it is not necessary/not encouraged to put the allocator as a field in the struct; this is typically only done when the struct has some internal fields/state that is allocated, and thus you need to keep track of the allocator to free those fields on `deinit`. 

Instead, if you really want to allow the user to customize their allocator, you can add the allocator as an argument to just this function.

Reply:
The difference between an arena allocator and a general purpose allocator is that an arena allocator is specifically designed for the use case where you might make many small allocations, but you do not want to/have to free any of those allocations individually. 

Instead, an arena makes allocations and turns the `free` function into a no-op, instead freeing _all_ of the allocations at once when the arena itself is `deinit`ed. 

This has the advantage of having significantly less overhead than a general purpose allocator that must keep track of extra state to reduce fragmentation, etc. 

In this case (only one allocation is made at the start), it doesn't really matter whether an arena is used or GPA is used, but I simply chose arena because it was simpler.

Reply:
Got a nice comments on allocator from Andrew in Ziggit: https://ziggit.dev/t/my-first-zig-library-looking-for-critics/5067/8. 

I feel this approach fits his 2nd point: _It’s part of the module, and it is not public_, correct? 

Is it a good idea to just leave the allocator completely to the users (the 3rd point)? 

Or simply get rid of the allocator at all, which makes cowsay just say a string, without the formatting feature?

Reply:
You can take in a user-provided allocator and wrap it in an arena! Or, you can actually implement this formatting with no allocations at all.